gnomad_qc/v4/annotations/insilico_predictors.py

```diff
@@ -89,6 +90,131 @@ def _load_cadd_raw(cadd_tsv) -> hl.Table:
     return ht
 
 
+def create_spliceai_grch38_ht(
+    snvs_path: str, indels_path: str, new_indels_path: str, header_file_path: str
+) -> hl.Table:
+    """
+    Create a Hail Table with SpliceAI scores for GRCh38.
+
+    SpliceAI scores are from the following resources:
+        - Precomputed SNVs: spliceai_scores.masked.snv.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - Precomputed indels: spliceai_scores.masked.indel.hg38.vcf.bgz,
+          downloaded from https://basespace.illumina.com/s/5u6ThOblecrh
+        - gnomAD v4 indels: gnomad_v4_new_indels.spliceai_masked.vcf.bgz,
+          computed on v4 by Illumina.
+
+    :param str snvs_path: Path to the precomputed SNVs.
+    :param str indels_path: Path to the precomputed indels.
+    :param str new_indels_path: Path to the gnomAD v4 indels.
+    :return: Hail Table with SpliceAI scores for GRCh38.
+    """
+
+    def import_spliceai_vcf(path: str) -> hl.Table:
+        """
+        Import SpliceAI VCF into Hail Table.
+
+        :param str path: Path to SpliceAI VCF.
+        :return: Hail MatrixTable with SpliceAI scores.
+        """
+        ht = hl.import_vcf(
+            path,
+            force_bgz=True,
+            header_file=header_file_path,
+            reference_genome="GRCh38",
+            contig_recoding=NO_CHR_TO_CHR_CONTIG_RECODING,
+            skip_invalid_loci=True,
+            min_partitions=1000,
+        ).rows()
+        return ht
+
+    logger.info("Importing vcf of SpliceAI scores into MT...")
+
+    spliceai_snvs = import_spliceai_vcf(snvs_path)
+    spliceai_indels = import_spliceai_vcf(indels_path)
+    spliceai_new_indels = import_spliceai_vcf(new_indels_path)
+
+    ht = spliceai_snvs.union(spliceai_indels).union(spliceai_new_indels)
+
+    logger.info("Exploding SpliceAI scores...")
+    # `explode` will eliminate rows with empty array, some variants don't have
+    # score might because they are located inside gene body, but the variants
+    # fall on multiple genes will extend the number of rows.
```

Comment:
```suggestion
    # `explode` because some variants fall on multiple genes and have a score per gene.
    # All rows without a SpliceAI score, an empty array, are removed through explode.
```